### PR TITLE
SAR-5476 - Play 2.6 Upgrade

### PR DESCRIPTION
**Pull request by HarrNM**
* [ ] I've included appropriate tests with any code I've added
* [ ] I've executed the acceptance test pack locally to ensure there are no regressions
* [x] I've added my code using logical, atomic commits, squashing as appropriate - including the Jira issue number in the commit message
* [x] I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: test/controllers/ConfirmationControllerSpec.scala [hunk 1]
```diff
@@ -25,44 +25,45 @@ import models.ConfirmationReferencesSuccessResponse
...
 import uk.gov.hmrc.http.HeaderCarrier
-import uk.gov.hmrc.play.test.WithFakeApplication
-import utils.Messages
 
-import scala.concurrent.Future
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.ExecutionContext.Implicits.global
 
-class ConfirmationControllerSpec extends SCRSSpec with CompanyDetailsFixture with WithFakeApplication with AuthBuilder {
+class øConfirmationControllerSpec(implicit val messages: Messages) extends SCRSSpec with CompanyDetailsFixture with GuiceOneAppPerSuite with AuthBuilder {
```
Comment (r-melvin) [inline test/controllers/ConfirmationControllerSpec.scala:40]: Can you remove the Scandinavian letter at the start of the class please?

File: app/services/AddressLookupConfigBuilderService.scala [hunk 2]
```diff
@@ -29,7 +29,7 @@ class AddressLookupConfigBuilderService @Inject()(appConfig: FrontendAppConfig)
   lazy val timeoutLength: Int = appConfig.timeoutInSeconds.toInt
   lazy val accessibilityFooterUrl: String = appConfig.accessibilityStatementUrl
 
-  def buildConfig(handbackLocation: Call, specificJourneyKey: String, lookupPageHeading: String, confirmPageHeading: String)(implicit messagesApi: MessagesApi): AlfJourneyConfig = {
+  def buildConfig(handbackLocation: Call, specificJourneyKey: String, lookupPageHeading: String, confirmPageHeading: String)(implicit messagesApi: MessagesApi, lang: Lang): AlfJourneyConfig = {
```
Comment (r-melvin) [inline app/services/AddressLookupConfigBuilderService.scala:32]: still unsure about lang

File: app/services/AddressLookupFrontendService.scala [hunk 3]
```diff
@@ -36,17 +35,17 @@ class AddressLookupFrontendService @Inject()(addressLookupFrontendConnector: Add
                                              addressLookupConfigBuilderService: AddressLookupConfigBuilderService,
                                              messagesApi: MessagesApi) {
 
-  lazy val addressLookupFrontendURL: String = frontendAppConfig.baseUrl("address-lookup-frontend")
+  lazy val addressLookupFrontendURL: String = frontendAppConfig.servicesConfig.baseUrl("address-lookup-frontend")
   lazy val companyRegistrationFrontendURL: String = frontendAppConfig.self
   lazy val timeoutInSeconds: Int = frontendAppConfig.timeoutInSeconds.toInt
 
-  def initialiseAlfJourney(handbackLocation: Call, specificJourneyKey: String, lookupPageHeading: String, confirmPageHeading: String)(implicit hc: HeaderCarrier): Future[String] = {
+  def initialiseAlfJourney(handbackLocation: Call, specificJourneyKey: String, lookupPageHeading: String, confirmPageHeading: String)(implicit hc: HeaderCarrier, lang: Lang): Future[String] = {
```
Comment (r-melvin) [inline app/services/AddressLookupFrontendService.scala:42]: as above

File: app/connectors/AddressLookupConnector.scala [hunk 4]
```diff
@@ -29,7 +29,7 @@ import scala.util.control.NoStackTrace
```
Comment (EviNikolaidou) [inline app/connectors/AddressLookupConnector.scala:29]: unused import play.api.libs.json.JsObject

File: app/forms/CancelForm.scala [hunk 5]
```diff
@@ -22,7 +22,7 @@ import play.api.i18n.Lang
```
Comment (EviNikolaidou) [inline app/forms/CancelForm.scala:22]: unsued import play.api.i18n.Lang

File: app/forms/CompanyContactForm.scala [hunk 6]
```diff
@@ -25,7 +25,7 @@ import utils.SCRSValidators.{companyContactDetailsValidation, emailValidation}
```
Comment (EviNikolaidou) [inline app/forms/CompanyContactForm.scala:25]: unused import play.api.i18n.Lang

File: app/forms/CompanyDetailsForm.scala [hunk 7]
```diff
@@ -22,7 +22,7 @@ import play.api.data.Forms._
 import play.api.i18n.Lang
```
Comment (EviNikolaidou) [inline app/forms/CompanyDetailsForm.scala:22]: unused import

File: app/forms/QuestionnaireForm.scala [hunk 8]
```diff
@@ -23,7 +23,7 @@ import play.api.i18n.Lang
```
Comment (EviNikolaidou) [inline app/forms/QuestionnaireForm.scala:23]: unused import play.api.i18n.Lang

File: app/forms/SubmissionForm.scala [hunk 9]
```diff
@@ -22,7 +22,7 @@ import play.api.data.Forms._
 import play.api.i18n.Lang
```
Comment (EviNikolaidou) [inline app/forms/SubmissionForm.scala:22]: unused import

File: app/models/NewAddress.scala [hunk 10]
```diff
@@ -124,8 +124,8 @@ object NewAddress {
       val auditRef = (json \ "auditRef").asOpt[String]
```
Comment (EviNikolaidou) [inline app/models/NewAddress.scala:125]: unused import play.api.data.validation.ValidationError

File: app/models/Validation.scala [hunk 11]
```diff
@@ -36,7 +36,7 @@ object Validation {
 
   def yyyymmddValidatorFmt = readToFmt(yyyymmddValidator)
 
-  def withFilter[A](fmt: Format[A], error: ValidationError)(f: (A) => Boolean): Format[A] = {
+  def withFilter[A](fmt: Format[A], error: JsonValidationError)(f: (A) => Boolean): Format[A] = {
```
Comment (EviNikolaidou) [inline app/models/Validation.scala:39]: unused import play.api.data.validation.ValidationError

File: app/models/handoff/CompanyNameHandOffInformation.scala [hunk 12]
```diff
@@ -21,6 +21,8 @@ import java.util.TimeZone
 import org.apache.commons.lang3.time.FastDateFormat
 import org.joda.time.DateTime
 import play.api.libs.json._
+import play.api.libs.json.JodaWrites._
+import play.api.libs.json.JodaReads._
```
Comment (EviNikolaidou) [inline app/models/handoff/CompanyNameHandOffInformation.scala:25]: intellij shows them unused ?

Comment (HarrNM) [reply app/models/handoff/CompanyNameHandOffInformation.scala:25]: Yeah - this is strange. It shows them as unused but without them there it fails to compile

**Pull request merged**